### PR TITLE
[jest]: fix ProjectConfig.transform type

### PR DESCRIPTION
**Pull request by Hotell**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://facebook.github.io/jest/docs/en/configuration.html#transform-object-string-string>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by Hotell** [comment]
ping

File: types/jest/index.d.ts [hunk 1]
```diff
@@ -863,7 +866,7 @@ declare namespace jest {
         testRunner: string;
         testURL: string;
         timers: 'real' | 'fake';
-        transform: Array<[string, Path]>;
```
Comment (ikatyang) [inline types/jest/index.d.ts]: This is how jest defined in their types: https://github.com/facebook/jest/blob/eb096f8b8d4a0db77fbce112fd758af7a6496764/types/Config.js#L270

I guess `InitialOptions` should be what you're looking for?
https://github.com/facebook/jest/blob/eb096f8b8d4a0db77fbce112fd758af7a6496764/types/Config.js#L157

**Review by ikatyang — CHANGES_REQUESTED** [review]

File: types/jest/index.d.ts (hunk 1 above)
Comment (Hotell) [reply types/jest/index.d.ts]: wow, that's super confusing naming in jest codebase indeed. But you are correct. 
Anyway we don't have `InitialOptions` interface here. So should I add it  ( mirror jest types in our TS types ) ? 

thanks

File: types/jest/index.d.ts [hunk 2]
```diff
@@ -801,6 +802,164 @@ declare namespace jest {
...
+        cacheDirectory: Path;
+        changedFilesWithAncestor: boolean;
+        clearMocks: boolean;
+        collectCoverage: boolean;
+        collectCoverageFrom: Maybe<string[]>;
+        coverageDirectory: Maybe<string>;
+        coveragePathIgnorePatterns: string[];
+        coverageReporters: string[];
+        coverageThreshold: Maybe<{global: {[key: string]: number}}>;
+        errorOnDeprecated: boolean;
```
Comment (ikatyang) [inline types/jest/index.d.ts]: This one is added in v23.1.0, we should either bump the version at Line 1 to `v23.1` or just remove it.

File: types/jest/index.d.ts [hunk 3]
```diff
@@ -801,6 +802,164 @@ declare namespace jest {
...
+        runner?: string;
+        runTestsByPath?: boolean;
+        scriptPreprocessor?: string;
+        setupFiles?: Path[];
+        setupTestFrameworkScriptFile?: Path;
+        silent?: boolean;
+        skipFilter?: boolean;
+        skipNodeResolution?: boolean;
+        snapshotSerializers?: Path[];
+        errorOnDeprecated?: boolean;
```
Comment (ikatyang) [inline types/jest/index.d.ts]: Same as above.

**Review by ikatyang — CHANGES_REQUESTED** [review]

File: types/jest/index.d.ts (hunk 3 above)
Comment (Hotell) [reply types/jest/index.d.ts]: kk bump sounds good. Also I've been thinking about extracting those configs in separate file, this one ambient index is getting too big -> hard to maintain. WDYT ?

Comment (ikatyang) [reply types/jest/index.d.ts]: Sounds good to me, and I think extracting all types mirrored from `jest` into separate files would be better, not just `Config.js`.

**Review by ikatyang — APPROVED** [review]

File: types/jest/index.d.ts (hunk 3 above)
Comment (Hotell) [reply types/jest/index.d.ts]: I'll do that in separate PR ✌️. Thx for reviewing and your time @ikatyang  ! 🍻

**Pull request merged**